src/main/java/org/skriptlang/skript/log/runtime/ErrorSource.java

```diff
@@ -0,0 +1,58 @@
+package org.skriptlang.skript.log.runtime;
+
+import ch.njol.skript.config.Node;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.lang.SyntaxElement;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A more versatile set of information about the source of an error.
+ * Aims to avoid relying specifically on {@link Node}s for information.
+ *
+ * @param syntaxType A string representing the type of syntax. See {@link SyntaxElement#getSyntaxTypeName()}.
+ * @param syntaxName The name of the syntax emitting the error.
+ * @param lineNumber The line number of the syntax emitting the error.
+ * @param lineText The raw code of the line.
+ * @param script The name of the script in which the line exists.
+ */
+public record ErrorSource(
+	String syntaxType,
+	String syntaxName,
+	int lineNumber,
+	String lineText,
+	String script
```

Comment:
Should we try to provide the Script instance?

Reply:
I believe that would require another method on producer for little benefit, since Node only provides access to Config, not Script.

Reply:
Fair enough - the consumer should be able to grab it if they need it.